### PR TITLE
Add sitemap, robots and manifest

### PR DESCRIPTION
**Pull request by wackerow**
## Description
- Adds `next/sitemap` package and `next-sitemap.config.js` config file
- Config also includes generation of `public/robots.txt` file (added to `.gitignore`)
- Adds `public/manifest.json` with necessary image assets

## Related Issue
https://www.notion.so/efdn/Add-robots-txt-and-sitemap-dbc564c8b7f240a1b5fd81187856bb54?pvs=4

### CODE REVIEW

File: next-sitemap.config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/** @type {import('next-sitemap').IConfig} */
+
+module.exports = {
+  siteUrl: process.env.SITE_URL || 'https://ethereum.org',
+  generateRobotsTxt: true,
```
Comment (pettinarip) [inline next-sitemap.config.js:7]: I see that we are indexing the 404 pages
https://deploy-preview-155--ethereum-org-fork.netlify.app/sitemap-0.xml

Maybe we could exclude those from the sitemap generation by using the `exclude` option.

**Review by pettinarip — COMMENTED** [review]
Looks good to me @wackerow 

Comparing it with the prod version, I see that we are changing the `changefreq` to `daily` and the `priority` to `0.5`. In prod we were using `monthly` & `0.7`. Not sure if this might have a negative impact or not, just wanted to call it out.

cc @nhsz not sure if you have any preference here

File: next-sitemap.config.js (hunk 1 above)
Comment (wackerow) [reply next-sitemap.config.js:7]: Yeah good call, looks like we're doing that on the Gatsby side
image

Will update

**Comment by wackerow** [comment]
> Comparing it with the prod version, I see that we are changing the changefreq to daily and the priority to 0.5. In prod we were using monthly & 0.7. Not sure if this might have a negative impact or not, just wanted to call it out.

Looks like in prod we're using `weekly`/`0.7` for _English_, and `monthly`/`0.5` for _non-English_. This plugin defaults to `daily`/`0.7`... I'll look into using a transform to add a conditional here to match parity with Gatsby site.

File: next-sitemap.config.js (hunk 1 above)
Comment (pettinarip) [reply next-sitemap.config.js:7]: @wackerow I still see some other 404 cases, like the ones for each lang
image

**Review by pettinarip — CHANGES_REQUESTED** [review]

File: next-sitemap.config.js (hunk 1 above)
Comment (pettinarip) [reply next-sitemap.config.js:7]: Fixed.

**Review by pettinarip — APPROVED** [review]

**Comment by wackerow** [comment]
Another option here that I left off was the `trailingSlash`... This has not been added to our `tsconfig` although we may want to consider; the canonical repo used this and if we plan to deploy a static export to IPFS I think this will be required. Doesn't have to block this PR, can discuss more.

**Review by nhsz — APPROVED** [review]
lgtm

**Pull request merged**